scripts/mixins/families/chigoe_pet.lua

```diff
@@ -0,0 +1,96 @@
+-- Chigoe(pet) family mixin
+
+require('scripts/globals/mixins')
+
+g_mixins = g_mixins or {}
+g_mixins.families = g_mixins.families or {}
+
+g_mixins.families.chigoe = function(chigoeMob)
+    -- Chigoes should spawn for the first 5 mobskills
+    chigoeMob:addListener('WEAPONSKILL_USE', 'MOB_SPAWN_CHIGOE', function(mobArg)
+        local chigoeCount = mobArg:getLocalVar('ChigoeCount')
+        local zone        = mobArg:getZone()
+
+        local chigoe = zone:insertDynamicEntity({
```

Comment:
Aren't chigoes in the base game regular entities with their own IDs?

Reply:
I suppose the way to check is to hop onto retail, get them to spawn (in B.Thickets) and see if they show any of these IDs:
```
INSERT INTO `mob_spawn_points` VALUES (16990289,'Chigoe','Chigoe',18,141.403,-18.403,299.224,69);
INSERT INTO `mob_spawn_points` VALUES (16990290,'Chigoe','Chigoe',18,140.637,-18.339,299.261,220);
INSERT INTO `mob_spawn_points` VALUES (16990291,'Chigoe','Chigoe',18,141.781,-18.189,298.570,78);
INSERT INTO `mob_spawn_points` VALUES (16990292,'Chigoe','Chigoe',18,159.025,-16.229,333.930,177);
INSERT INTO `mob_spawn_points` VALUES (16990293,'Chigoe','Chigoe',18,156.374,-15.863,345.795,118);
```

Reply:
Better yet, a combined approach where you look up the IDs of the chigoes in mob ID area and see if they're spawned or not, and then fall back to using DEs

Reply:
Do what happens on retail. If they're DE's, then use DE's, if they're mob-space mobs that can't spawn if all their IDs are in use - do that.

"Better than retail" can come later.